nomos-services/data-availability/indexer/src/storage/adapters/rocksdb.rs

```diff
@@ -0,0 +1,117 @@
+use tokio_stream::wrappers::ReceiverStream;
+
+use crate::storage::DaStorageAdapter;
+
+const DA_KEY_PREFIX: &str = "da";
+
+pub struct RocksAdapter<S, V>
+where
+    S: StorageSerde + Send + Sync + 'static,
+    V: VID + Metadata + Send + Sync,
+{
+    storage_relay: OutboundRelay<StorageMsg<RocksBackend<S>>>,
+    _vid: PhantomData<V>,
+}
+
+#[async_trait::async_trait]
+impl<S, V> DaStorageAdapter for RocksAdapter<S, V>
+where
+    S: StorageSerde + Send + Sync + 'static,
+    V: VID<CertificateId = [u8; 32]> + Metadata + Send + Sync,
+    V::Index: AsRef<[u8]> + Next + Clone + PartialOrd + Send + Sync + 'static,
+    V::AppId: AsRef<[u8]> + Send + Sync,
+{
+    type Backend = RocksBackend<S>;
+    type Blob = Bytes;
+    type VID = V;
+
+    async fn new(
+        storage_relay: OutboundRelay<<StorageService<Self::Backend> as ServiceData>::Message>,
+    ) -> Self {
+        Self {
+            storage_relay,
+            _vid: PhantomData,
+        }
+    }
+
+    async fn add_index(&self, vid: Self::VID) -> bool {
+        let (app_id, idx) = vid.metadata();
+        let key = meta_to_key_bytes(app_id, idx);
+        let value = Bytes::from(vid.certificate_id().to_vec());
+
+        self.storage_relay
+            .send(StorageMsg::<Self::Backend>::Store { key, value })
+            .await
+            .expect("Storage backend should be ready");
+
+        false
+    }
+
+    async fn get_range_stream(
+        &self,
+        app_id: <Self::VID as Metadata>::AppId,
+        index_range: Range<<Self::VID as Metadata>::Index>,
+    ) -> Box<dyn Stream<Item = (<Self::VID as Metadata>::Index, Option<Bytes>)> + Unpin + Send>
+    {
+        let futures = FuturesUnordered::new();
+
+        let mut current_index = index_range.start.clone();
+        while current_index < index_range.end {
+            let idx = current_index.clone();
+
+            let (reply_tx, reply_rx) = tokio::sync::oneshot::channel();
+            let key = meta_to_key_bytes(app_id.as_ref(), current_index.as_ref());
+
+            self.storage_relay
+                .send(StorageMsg::Load {
+                    key,
+                    reply_channel: reply_tx,
+                })
+                .await
+                .expect("Failed to send load request to storage relay");
+
+            futures.push(async move {
+                match reply_rx.await {
+                    Ok(Some(blob)) => (idx, Some(blob)),
+                    Ok(None) => (idx, None),
+                    Err(_) => {
+                        eprintln!("Failed to receive storage response");
+                        (idx, None)
+                    }
+                }
+            });
+
+            current_index = current_index.next();
+        }
+
+        Box::new(futures)
+    }
+}
+
+fn meta_to_key_bytes(app_id: impl AsRef<[u8]>, idx: impl AsRef<[u8]>) -> Bytes {
+    let mut buffer = BytesMut::with_capacity(42 + app_id.as_ref().len() + idx.as_ref().len());
```

Comment:
`42` better as a constant.

Reply:
`meta_to_key_bytes` was changed to `key_bytes`, constant removed:
https://github.com/logos-co/nomos-node/pull/644/files#diff-bcd2717fdf8fd2ef24063fd29eae1b9bd1f96f1d75fb41982217f136d211a254R151-R158

You've probably reviewed commit by commit, sorry for not keeping the commits domain specific.

Reply:
@bacv I think I was having some issues and maybe I review an older version or who knows. Looks like everything is working well now.